docs/source/blocks/block-style-wrapper.md

```diff
@@ -293,89 +172,165 @@ It will generate these values for the `StyleWrapper` to use:
 
 ```js
-  schema.properties.styles.schema.fieldsets[0].fields = [
-    ...schema.properties.styles.schema.fieldsets[0].fields,
-    'theme:noprefix',
-  ];
-
-  schema.properties.styles.schema.properties['theme:noprefix'] = {
-    widget: 'color_picker',
-    title: "Theme",
-    colors,
-    default: defaultBGColor,
-  };
+schema.properties.styles.schema.fieldsets[0].fields = [
+  ...schema.properties.styles.schema.fieldsets[0].fields,
+  'theme:noprefix',
+];
+
+schema.properties.styles.schema.properties['theme:noprefix'] = {
+  widget: 'color_picker',
+  title: 'Theme',
+  colors,
+  default: defaultBGColor,
+};
 ```
 
-It will generate these values for the `StyleWrapper` to use:
+This yields inline variables without the prefix:
 
-```js
+```html
+<div class="block teaser" style="--background-color: #222; --font-color: white">
+  ...
+</div>
+```
+
+(block-style-wrapper-generated-class-names-label)=
+## Generated class names
+
+The wrapper can generate class names from the `styles` object.
+This is useful when mapping discrete, curated style tokens to theme classes.
+
+Class names are built by concatenating key–value pairs with `--` and prefixing with `has--`.
+One level of nesting is supported.
+
+Given:
+
+```json
 {
   "styles": {
-    "theme:noprefix": {
-      "--background-color": "#222",
-      "--font-color": "white"
+    "backgroundColor": "#ee22ee",
+    "myCustomStyleField": "red",
+    "myCustom2StyleField": {
+      "color": "black",
+      "gradient": "MyGradient"
     }
   }
 }
 ```
 
-This will yield the resultant markup without a prefix.
+Resulting classes:
 
 ```html
-<div class="block teaser" style="--background-color: #222, --font-color: white">
- ...
-</div>
+<div class="has--backgroundColor--ee22ee has--myCustomStyleField--red has--myCustom2StyleField--color--black has--myCustom2StyleField--gradient--MyGradient">
 ```
 
-## Align class injection
+Pass the injected `className` to the root element in your view component:
+
+```jsx
+const BlockView = (props) => (
+  <div className={props.className}>
+    {/* Block's view component code */}
+  </div>
+);
+```
+
+### Customize generated class names
 
-There is an automatic class name injection happening at the same time the style wrapper class names injection.
-The `data.align` is also injected directly.
-This is in place to help properly position the block in the current layout and play well with legacy CSS and block layout.
-This might be replaced in the future by the style wrapper class names injection.
+You can customize class generation with converters in `config.settings.styleClassNameConverters` by suffixing field names with the converter name.
+For example:
 
-Each block in the Block Engine has a main wrapper with an automatic class name `block-editor-<block_id> <block_align>`, as shown in the following example:
+```
+{
+  "styles": {
+    "theme:noprefix": "primary",
+    "inverted:bool": true
+  }
+}
+```
+
+Generates: `primary inverted`.
```

Comment:
Cosmetically and structurally, it looks funky. https://volto--7405.org.readthedocs.build/7405/blocks/block-style-wrapper.html#customize-generated-class-names

````suggestion
Generates:

```css
primary inverted
```
````